rmw_cyclonedds_cpp/src/serdata.cpp

```diff
@@ -465,9 +465,9 @@ bool sertopic_rmw_equal(
 uint32_t sertopic_rmw_hash(const struct ddsi_sertopic * tpcmn)
 {
   const struct sertopic_rmw * tp = static_cast<const struct sertopic_rmw *>(tpcmn);
-  uint32_t h2 = std::hash<bool>{} (tp->is_request_header);
-  uint32_t h1 = std::hash<std::string>{} (std::string(tp->type_support.typesupport_identifier_));
-  return h1 ^ h2;
+  size_t h2 = std::hash<bool>{} (tp->is_request_header);
+  size_t h1 = std::hash<std::string>{} (std::string(tp->type_support.typesupport_identifier_));
+  return static_cast<uint32_t>(h1 ^ h2);
```

Comment:
Nitpick: first doing the operator and then throwing away part of the result seems unnecessary. How about casting both arguments (or the variables above) and then only do the operator for the necessary bits? (Not that it will matter performance wise.)

Reply:
Done in 02d0e18